draft-specifications/Model.md

```diff
@@ -526,69 +472,56 @@ local peer, the sent and received values will have the same type.
+Values of different OCapN types are not Equal.
+Values of the same OCapN type may be Equal under conditions specific to their
+shared type.
+All OCapN types maintain pass invariant Equality except Promise.
 
-> Equality does not guarantee a unique identity for all equivalent values.
+All OCapN implementations, regardless of language, must model all OCapN types
+such that Equality for the in-language representation of values is consistent
+with the Equality and Pass Invariance of Equality for their corresponding OCapN
+model type.
 
-Any pair of values of different types are not equal or equivalent regardless of
-their values.
+A type holds Equality invariant over passage between OCapN peers because any
+pair of local values that are Equal can be passed to another and the remote
+values will also be Equal.
 
-> Only values of the same type can be equivalent, per the equivalence relation
-> specified for their common type.
+It follows that, for any local value that is sent to a remote peer and then returned,
+the sent and received values will be Equal.
 
-OCapN holds equality invariant for all [Atoms](#atom) and
-[Containers](#container) passed between peers.
+> OCapN Equal values may be locally distinguishable by oother operators
+> depending on the language implementation and type.
 
 > In JavaScript, none of `==`, `===`, or `Object.is` are sufficient to compare
-> equality of values, but a composite operator that takes the OCapN type for
-> every value into account.
+> Equality of values, which instead must take into account the OCapN type for
+> each value.
 > For example, Float64 can be compared by `Object.is`, which preserves the
-> distinction between 0 and -0 and non-equivalence of any NaN to any other NaN,
-
-## Pass Invariant Identity
-
-Values of a type have an identity if a value of that type is identical to
-itself and no other value.
-A type holds identity invariant over passage between OCapN peers if values of that
-type can be passed from a local peer to a remote peer, then returned to the local peer,
-and the value that was sent is identical the the value received _and identical
-to no other value_.
-
-> The operators necessary for checking identify may vary for each language and
-> even for types in one language.
-> Pass Invariant Identity obliges OCapN to track a value's identity until the value
-> becomes unreachable.
-
-OCapN holds identity invariant only for values of type [Target](#target).
-
-OCapN does not hold identity invariant for values of type [Promise](#promise).
-
-OCapN does not guarantee identity is invariant for any values of [Atom](#atom)
-or [Container](#container) types.
+> distinction between 0 and -0 and lack of distinction between any two NaN
+> values.
+> `Object.is` is also sufficient for Target provided that the OCapN
+> implementation holds invariant a single object identity for every Target.
+> The JavaScript ArrayBuffer representation of an OCapN ByteArray must be
+> compared byte-for-byte.
```

Comment:
Good! 

Also, `Object.is` doesn't work on containers because container equality is based on recursive equality of the contents.

Reply:
Added.